packages/odata-service-inquirer/src/utils/index.ts

```diff
@@ -0,0 +1,64 @@
+import { isAppStudio } from '@sap-ux/btp-utils';
+import type { TelemetryEvent, TelemetryProperties, ToolsSuiteTelemetryClient } from '@sap-ux/telemetry';
+import { SampleRate } from '@sap-ux/telemetry';
+import osName from 'os-name';
+import { PLATFORMS } from '../types';
+
+const osVersionName = osName();
+
+/**
+ * Determine if the current prompting environment is cli or a hosted extension (app studio or vscode).
+ *
+ * @returns the platform name and technical name
+ */
+export function getPlatform(): { name: string; technical: string } {
+    if ((process.mainModule && process.mainModule.filename.includes('yo')) ?? process.stdin.isTTY) {
+        return PLATFORMS.CLI;
+    } else {
+        return isAppStudio() ? PLATFORMS.SBAS : PLATFORMS.VSCODE;
+    }
+}
+
+let telemetryClient: ToolsSuiteTelemetryClient | undefined;
+
+/**
+ * Set the telemetry client.
+ *
+ * @param toolsSuiteTelemetryClient
+ */
+export function setTelemetryClient(toolsSuiteTelemetryClient: ToolsSuiteTelemetryClient | undefined): void {
+    telemetryClient = toolsSuiteTelemetryClient;
+}
+
+/**
+ * Send telemetry event.
+ *
+ * @param eventName the name of the telemetry event
+ * @param telemetryData the telemetry values to report
+ */
+export function sendTelemetryEvent(eventName: string, telemetryData: TelemetryProperties): void {
```

Comment:
Probably stating the obvious but we'll have to move this somewhere?

Reply:
Not currently used anywhere else so Im going to leave it for now.